Yoast/javascript
# LINGO-685 Make researcher argument of assessor

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Makes the researcher an argument of assessor.

## Relevant technical choices:

* Pre-refactor, the `assess` method of the assessor would create a new researcher if no researcher was provided in the options when constructing the assessor. Now, we don't have one general researcher anymore. The assessor always needs to be constructed with an appropriate language-specific researcher. That's why `researcher` now is made a required argument when constructing the assessor. Not providing a researcher when constructing the assessor will result in an error.
* This branch refers to an incorrect issue number, so I added the correct issue number in the PR title. 
* The empty premium-configuration branch with the same name **should be deleted** once this branch is merged.
* Removed the locale option that was still passed to some assessors. The assessors don't use the locale anymore, so it's not necessary to pass it in the options. I think the locale might be removed from more parts of the web worker. But it's not clear if it can be totally removed. I made a [technical debt issue](https://yoast.atlassian.net/browse/LINGO-696) to investigate this.

## Test instructions

This PR can be tested by following these steps:

* Confirm that all specs pass (currently failing unrelated specs: full texts tests for Dutch and Portuguese)
* In the plugin (testing this branch together with the [associated plugin branch](https://github.com/Yoast/wordpress-seo/pull/16731)):
  * Open a post. Do a smoke test confirming that you get analysis results. Also confirm that you don't get any errors in the console. (There are some warnings at the moment. These are also present on `trunk`. You can ignore those when testing this PR.)
  * Open a taxonomy page and do the same.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-668
